lib/puppet/provider/package/chocolatey.rb

```diff
@@ -51,6 +51,25 @@ def is_compiled_choco?
     self.class.is_compiled_choco?
   end
 
+  def is_usepackageexitcodes_feature_enabled?
+    PuppetX::Chocolatey::ChocolateyCommon.set_env_chocolateyinstall
+
+    choco_config = PuppetX::Chocolatey::ChocolateyCommon.choco_config_file
+    raise Puppet::ResourceError, "Config file not found for Chocolatey. Please make sure you have Chocolatey installed." if choco_config.nil?
+    raise Puppet::ResourceError, "An install was detected, but was unable to locate config file at #{choco_config}." unless PuppetX::Chocolatey::ChocolateyCommon.file_exists?(choco_config)
+
+    config = REXML::Document.new File.read(choco_config)
+    feature_tags = config.elements.to_a('//feature').select do |feature|
+      feature.attributes['name'].downcase == 'usepackageexitcodes'
+    end
+
+    return false if feature_tags.empty?
+    feature = feature_tags.first
+
+    return true if feature.attributes['enabled'].downcase == 'true'
```

Comment:
This should check if it was explicitly enabled - `<feature name="usePackageExitCodes" enabled="true" setExplicitly="false" description="[..snip..]" />`

Otherwise this is likely to break all folks - the feature is turned on by default.